test/transceivercontroller/DefaultTransceiverController.test.ts

```diff
@@ -590,4 +588,189 @@ describe('DefaultTransceiverController', () => {
+      tc.setPeer(peer);
+      tc.setupLocalTransceivers();
+      const newVideoTrack = new MediaStreamTrack();
+      await tc.setVideoInput(newVideoTrack);
+
+      const localSender = peer.getTransceivers()[1].sender;
+      const setParamSpy = sinon.spy(localSender, 'setParameters');
+      await tc.setEncodingParameters(
+        new Map<string, RTCRtpEncodingParameters>().set('video', {
+          rid: 'video',
+          scaleResolutionDownBy: 1,
+          maxBitrate: 1400000,
+        })
+      );
+
+      expect(setParamSpy.calledOnce).to.be.true;
+      let params = localSender.getParameters();
+      expect(params.encodings.length).to.be.equal(1);
+      expect(params.encodings[0].maxBitrate).to.be.equal(1400000);
+      expect(params.encodings[0].scaleResolutionDownBy).to.be.equal(1);
+
+      await tc.setEncodingParameters(
+        new Map<string, RTCRtpEncodingParameters>().set('video', {
+          rid: 'video2',
+          scaleResolutionDownBy: 2,
+          maxBitrate: 600000,
+        })
+      );
+      expect(setParamSpy.calledTwice).to.be.true;
+      params = localSender.getParameters();
+      expect(params.encodings.length).to.be.equal(1);
+      expect(params.encodings[0].maxBitrate).to.be.equal(1400000);
+      expect(params.encodings[0].scaleResolutionDownBy).to.be.equal(1);
+
+      setParamSpy.restore();
+    });
+
+    it('Does not set codecPayloadType', async () => {
+      const peer: RTCPeerConnection = new RTCPeerConnection();
+      tc.setPeer(peer);
+      tc.setupLocalTransceivers();
+      const newVideoTrack = new MediaStreamTrack();
+      await tc.setVideoInput(newVideoTrack);
+
+      const localSender = peer.getTransceivers()[1].sender;
+      const setParamSpy = sinon.spy(localSender, 'setParameters');
+      await tc.setEncodingParameters(
+        new Map<string, RTCRtpEncodingParameters>().set('video', {
+          rid: 'video',
+          codecPayloadType: 1,
+          scaleResolutionDownBy: 1,
+          maxBitrate: 1400000,
+        })
+      );
+
+      expect(setParamSpy.calledOnce).to.be.true;
+      let params = localSender.getParameters();
+      expect(params.encodings.length).to.be.equal(1);
+      expect(params.encodings[0].maxBitrate).to.be.equal(1400000);
+      expect(params.encodings[0].scaleResolutionDownBy).to.be.equal(1);
+
+      await tc.setEncodingParameters(
+        new Map<string, RTCRtpEncodingParameters>().set('video', {
+          rid: 'video',
+          codecPayloadType: 2,
+          scaleResolutionDownBy: 2,
+          maxBitrate: 600000,
+        })
+      );
+      expect(setParamSpy.calledTwice).to.be.true;
+      params = localSender.getParameters();
+      expect(params.encodings.length).to.be.equal(1);
+      expect(params.encodings[0].maxBitrate).to.be.equal(600000);
+      expect(params.encodings[0].scaleResolutionDownBy).to.be.equal(2);
+      expect(params.encodings[0].codecPayloadType).to.be.equal(1);
+
+      setParamSpy.restore();
```

Comment:
FYI, there's `afterEach` which might help you kill some cleanup boilerplate.

Reply:
Move it out to `afterEach`